librz/core/cprint.c

```diff
@@ -30,4 +30,137 @@ RZ_API char *rz_core_print_string_c_cpp(RzCore *core) {
 	}
 	rz_strbuf_append(sb, "\";");
 	return rz_strbuf_drain(sb);
+}
+
+/**
+ * \brief Get the hexpair of the assembly
+ * \param core RzCore
+ * \param assembly assembly
+ * \return a string containing the hexpair of the assembly
+ */
+RZ_API RZ_OWN char *rz_core_hex_of_assembly(RzCore *core, const char *assembly) {
+	RzStrBuf *buf = rz_strbuf_new("");
+	if (!buf) {
+		RZ_LOG_ERROR("Fail to allocate memory\n");
+		return NULL;
+	}
+	rz_asm_set_pc(core->rasm, core->offset);
+	RzAsmCode *acode = rz_asm_massemble(core->rasm, assembly);
+	if (!acode) {
+		RZ_LOG_ERROR("Fail to assemble by rz_asm_massemble()\n");
+		rz_strbuf_free(buf);
+		return NULL;
+	}
+	for (int i = 0; i < acode->len; i++) {
+		ut8 b = acode->bytes[i]; // core->print->big_endian? (bytes - 1 - i): i ];
+		rz_strbuf_appendf(buf, "%02x", b);
+	}
+	rz_asm_code_free(acode);
+	return rz_strbuf_drain(buf);
+}
+
+/**
+ * \brief Get the esil of the assembly
+ * \param core RzCore
+ * \param assembly assembly
+ * \return a string containing the esil of the assembly
+ */
+RZ_API RZ_OWN char *rz_core_esil_of_assembly(RzCore *core, const char *assembly) {
+	RzStrBuf *buf = rz_strbuf_new("");
+	if (!buf) {
+		RZ_LOG_ERROR("Fail to allocate memory\n");
+		return NULL;
+	}
+	rz_asm_set_pc(core->rasm, core->offset);
+	RzAsmCode *acode = rz_asm_massemble(core->rasm, assembly);
+	if (!acode) {
+		RZ_LOG_ERROR("Fail to assemble by rz_asm_massemble()\n");
+		rz_strbuf_free(buf);
+		return NULL;
+	}
+	int printed = 0, bufsz = acode->len;
+	RzAnalysisOp aop = { 0 };
+	while (printed < bufsz) {
+		aop.size = 0;
+		if (rz_analysis_op(core->analysis, &aop, core->offset,
+			    (const ut8 *)acode->bytes + printed, bufsz - printed, RZ_ANALYSIS_OP_MASK_ESIL) <= 0 ||
+			aop.size < 1) {
+			RZ_LOG_ERROR("Cannot decode instruction\n");
+			rz_analysis_op_fini(&aop);
+			rz_strbuf_free(buf);
+			rz_asm_code_free(acode);
+			return NULL;
+		}
+		rz_strbuf_appendf(buf, "%s\n", RZ_STRBUF_SAFEGET(&aop.esil));
+		printed += aop.size;
+		rz_analysis_op_fini(&aop);
+	}
+	rz_asm_code_free(acode);
+	return rz_strbuf_drain(buf);
+}
+
+/**
+ * \brief Get the assembly of the hexstr
+ * \param core RzCore
+ * \param hexstr hexstr
+ * \return a string containing the assembly of the hexstr
+ */
+RZ_API RZ_OWN char *rz_core_assembly_of_hex(RzCore *core, const char *hexstr) {
+	RzStrBuf *buf = rz_strbuf_new("");
+	if (!buf) {
+		RZ_LOG_ERROR("Fail to allocate memory\n");
+		return NULL;
+	}
+	rz_asm_set_pc(core->rasm, core->offset);
+	bool is_pseudo = rz_config_get_i(core->config, "asm.pseudo");
+	RzAsmCode *acode = rz_asm_mdisassemble_hexstr(core->rasm, is_pseudo ? core->parser : NULL, hexstr);
```

Comment:
if you make `rz_core_assembly_of_hex` accept `ut8 *hex, int len` you can probably call the regular rz_asm_mdissassemble here and then remove completely the `rz_asm_mdisassemble_hexstr` function.